native/common/jp_field.cpp

```diff
@@ -105,7 +105,7 @@ void JPField::setStaticAttribute(HostRef* val)
 	{
 		stringstream err;
 		err << "unable to convert to " << type->getName().getSimpleName();
-		RAISE(JPypeException, err.str().c_str());
+		JPEnv::getHost()->setTypeError( err.str().c_str());

```

Comment:
was the exception not propagated correctly or why is this change needed?

Reply:
The previous exception was just a generic RuntimeError.  I believe that it should properly be TypeError as per the python description.   We could live with either way but when setting up the testing framework I was left scratching my head as to what exception type should have been.  I originally wrote it as TypeError then found it failed because it was actually a RuntimeError.  I thought it best to make it clear the difference between trying to shove the wrong type of data into a strongly typed language versus trying to put the right type of data into a read only field.